rules/coding-style/src/Rector/Property/UnderscoreToPascalCasePropertyNameRector.php

```diff
@@ -71,32 +71,39 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        if (Strings::startsWith($nodeName, '_')) {
+        /** @var string $class */
+        $class = $node->getAttribute(AttributeKey::CLASS_NAME);
+        // properties are accessed via magic, nothing we can do
+        if (method_exists($class, '__set') || method_exists($class, '__get')) {
             return null;
         }
 
-        if (! Strings::contains($nodeName, '_')) {
+        // @see https://regex101.com/r/dXc89K/1
+        // matches if underscore is present but excluding first underscore
+        if (! Strings::match($nodeName, '#(?<!^)\_#')) {
             return null;
         }
 
         $camelCaseName = $this->createPascalName($nodeName, $node);
```

Comment:
Probably incorrect name on the left or on the right :D

```diff
-$camelCaseName = $this->createPascalName($nodeName, $node);
+$pascalCaseName = $this->createPascalName($nodeName, $node);
```

or 

```diff
-$camelCaseName = $this->createPascalName($nodeName, $node);
+$camelCaseName = $this->createCamelCaseName($nodeName, $node);
```

Reply:
:D it was in original rule and I didn't pay attention too much to this name.

Reply:
No troubles. I just noticed it ;)

Reply:
There should be a PHPStan/Rector rule for that, right?

Reply:
It's gonna be too strict then! :D